### ISSUE TITLE
Do not post a comment for unrelated errors

### ISSUE THREAD

**Issue by Kobzol** [rust-lang/bors#124]
After every comment on a PR, the bot tries to load the PR information from GitHub. However, this can fail, and in that case the bot starts spamming the PR with [error messages](https://github.com/rust-lang/rust/pull/128861#issuecomment-2286217324). The bot should post a comment when it encounters an error during its operation, but not when it just deals with a comment that is completely unrelated to it. We should first find out whether the comment actually contains a bors command, and if not, we should just always ignore it.

I originally included the PR info loading logic unconditionally to reduce code duplication (so that we always load the PR regardless of the event), but it was a bad idea. We are also needlessly spamming GitHub now, even though most comments are unrelated to bors.

**Comment by Sakib25800** [comment]
Would this be as simple as bailing when there's no commands? 

```rust
async fn handle_comment(
    repo: Arc<RepositoryState>,
    database: Arc<PgDbClient>,
    ctx: Arc<BorsContext>,
    comment: PullRequestComment,
) -> anyhow::Result<()> {
    let pr_number = comment.pr_number;
    let commands = ctx.parser.parse_commands(&comment.text);

    // Bail if no commands 
    if commands.is_empty() {
        return Ok(());
    }

    tracing::debug!("Commands: {commands:?}");
    tracing::trace!("Text: {}", comment.text);

    let pull_request = ...
```

**Comment by Kobzol** [comment]
Yes, that should work! Plus a test where we purposely make `get_pull_request` fail, and check that bors doesn't send any error comment to the PR if the comment doesn't contain any bors commands.
